File: admin/src/main/java/com/imstargg/admin/domain/StarPower.java
package com.imstargg.admin.domain;

import com.imstargg.storage.db.core.MessageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BrawlStarsImageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.StarPowerCollectionEntity;
import jakarta.annotation.Nullable;

import java.util.List;

public record StarPower(
        StarPowerCollectionEntity entity,
        List<MessageCollectionEntity> names,
        @Nullable BrawlStarsImageCollectionEntity image
) {
}

File: admin/imstargg-admin-frontend/src/model/StarPower.ts
import {MessageEntity} from "@/model/entity/MessageEntity";
import {StarPowerEntity} from "@/model/entity/StarPowerEntity";
import { BrawlStarsImageEntity } from "./entity/BrawlStarsImageEntity";

export interface StarPower {
  entity: StarPowerEntity;
  names: MessageEntity[];
  image: BrawlStarsImageEntity | null;
}